### PR TITLE
MAINT: testing syntax

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses pysat/pysat#1053, pysat/pysat#1042, pysat/pysat#1048

- Updates testing syntax to maintain pytest standards. 
- Updates Github Actions syntax
- Removes nose requirements
- restores changelog from previous version
- Adds RC workflow

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Via Github Actions

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

 If this is a release PR, replace the first item of the above checklist with the
 release checklist on the pysat wiki:
 https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,4 +13,21 @@ This project adheres to [Semantic Versioning](https://semver.org/).
...
+  * Added pull request templates and other GitHub project documentation.
+  * Switched Windows installation instructions to favor installing WSL.
+* Bug Fix
+  * Improved builds for newer compilers.
+  * Replaces uninterpretable characters with '*' so data loading may continue.
+* Maintenance
+  * Adopted latest pysat development standards.
+  * Shifted from TravisCI to GitHub Actions for online testing.
+  * Adopted setup.cfg
+  * Improved PEP8 compliance
```
Comment (aburrell) [inline CHANGELOG.md:39]: This contains a lot more changes than I see in this pull request.

Comment (jklenzing) [reply CHANGELOG.md:39]: During a previous pull, the changelog was apparently deleted for previous versions.  restores based on release notes.

Comment (jklenzing) [reply CHANGELOG.md:39]: merged changes from main into develop-3.  This bit still shows up as a duplicate since it's a separate commit.

**Comment by jklenzing** [comment]
Added a flake-8 cap to this branch since the latest version does not like how we invoke the ignore rules.  Documented as part of standards upgrade in #43.

File: .github/workflows/pysat_rc.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Test with latest pysat RC
+
+on: [push]
```
Comment (jklenzing) [inline .github/workflows/pysat_rc.yml]:
```suggestion
on: [workflow_dispatch]
```

**Comment by jklenzing** [comment]
Went ahead and added the RC check to close out issues at pysat.  Results of test are here: https://github.com/pysat/pysatCDF/actions/runs/3594442005

Added #44 to capture the updating Github Actions to include windows and mac tests.  Probably best for these to be implemented after or during meson upgrades to the install procedure.

**Review by aburrell — APPROVED** [review]

**Comment by rstoneback** [comment]
Thanks @jklenzing and @aburrell!

**Pull request merged**